### DISCUSSION TITLE
How to use spack-stack to create python module/environments etc

### DISCUSSION THREAD

**Discussion by JessicaMeixner-NOAA** [JCSDA/spack-stack#858] [Q&A]
In hpc-stack, we could create a yml/conda recipe with python modules that are desired and create a module that could then be used for that set of python modules.    Is there a way to do this with spack-stack?  I've looked at the documentation and don't see how to do this.

**Comment by climbfuji** [comment]
This is what we do at JCSDA with JEDI (including EWOK-Skylab workflow, that has a ton of dependencies):

1. The usual basic spack load: `module use ... && module load stack-compiler && module load stack-mpi && module load stack-python`
2. Load JEDI modules that contain a number of python packages (can also load them individually): `module load jedi-fv3-env soca-env ewok-env`
3. Create virtual environment and install missing packages (make sure that `which python3` points to the spack-built python): `python3 -m venv venv && source venv/bin/activate && python3 -m pip install missing-package`. This will use existing python packages in spack-stack environment as dependencies for packages that are getting installed.
